src/V3Width.cpp

```diff
@@ -2643,14 +2644,51 @@ class WidthVisitor final : public VNVisitor {
 
         iterateCheck(nodep, "Dist expression", nodep->exprp(), CONTEXT_DET, FINAL, subDTypep,
                      EXTEND_EXP);
+        AstNodeExpr* newp = nullptr;
         for (AstDistItem *nextip, *itemp = nodep->itemsp(); itemp; itemp = nextip) {
             nextip
                 = VN_AS(itemp->nextp(), DistItem);  // iterate may cause the node to get replaced
             iterateCheck(nodep, "Dist Item", itemp, CONTEXT_DET, FINAL, subDTypep, EXTEND_EXP);
+            AstNodeExpr* inewp = nullptr;
+            const AstNodeDType* const itemDtp = itemp->dtypep()->skipRefp();
+            if (AstInsideRange* const irangep = VN_CAST(itemp->rangep(), InsideRange)) {
+                inewp = irangep->newAndFromInside(nodep->exprp(), irangep->lhsp()->unlinkFrBack(),
+                                                  irangep->rhsp()->unlinkFrBack());
+            } else if (VN_IS(itemDtp, UnpackArrayDType) || VN_IS(itemDtp, DynArrayDType)
+                       || VN_IS(itemDtp, QueueDType)) {
+                // Unsupported in parameters
+                AstNodeExpr* exprp = nodep->exprp()->cloneTreePure(true);
+                inewp = new AstCMethodHard{itemp->fileline(), itemp->rangep()->unlinkFrBack(),
+                                           "inside", exprp};
+                iterateCheckTyped(nodep, "inside value", exprp, itemDtp->subDTypep(), BOTH);
+                VL_DANGLING(exprp);  // Might have been replaced
+                inewp->dtypeSetBit();
+                inewp->didWidth(true);
+            } else if (VN_IS(itemDtp, AssocArrayDType)) {
+                nodep->v3error("Inside operator not specified on associative arrays "
+                               "(IEEE 1800-2023 11.4.13)");
+                continue;
```

Comment:
Make message slightly different from the other similar message, and need test for this.

Reply:
Moved to a common function, so it can probably stay the same?

Reply:
Still need a test for code branch coverage. It's easiervto know it's hit if message is different, but alternatively you can use code coverage or a debug message to check when you write the test.

Reply:
Not sure what you mean, there is now only one function throwing the same error message, regardless of whether this is a `dist` or `inside` expression. They share a lot of semantics, so I decided that one separate function to handle their common logic would be better. Should I add an extra test that tests array ranges in `dist` constraints? (This would probably need #5413 first, but I might be wrong.)

Reply:
Sorry I was misreading the change, you're ok with the error as-is then.

>Should I add an extra test that tests array ranges in dist constraints? (This would probably need https://github.com/verilator/verilator/pull/5413 first, but I might be wrong.)

Yes, please/thanks.